aea/helpers/base.py

```diff
@@ -68,3 +70,13 @@ def locate(path):
         except AttributeError:
             return None
     return object
+
+
+def generate_fingerprint(author: str, package_name: str, version: str, nonce: Optional[int] = None) -> str:
+    """Generate a unique id for the package."""
+    import hashlib
+    string_for_hash = "".join([author, package_name, version, str(nonce)])
```

Comment:
If the nonce = None, we should not convert it to the string, rather just have the first three